src/core/shape_inference/include/string_tensor_unpack_shape_inference.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+
+#include "openvino/op/string_tensor_unpack.hpp"
+#include "shape_infer_type_utils.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+namespace util {
+static Tensor get_string_tensor(const Node* op, size_t idx, const ITensorAccessor& tensor_accessor) {
+    if (auto t = tensor_accessor(idx)) {
+        return t;
+    } else {
+        const auto& constant = as_type_ptr<opset1::Constant>(op->get_input_node_shared_ptr(idx));
+        NODE_VALIDATION_CHECK(op, constant != nullptr, "Static shape inference lacks constant data on port ", idx);
```

Comment:
The check can be removed outside, should be only used for static shapes.

Reply:
Done.